clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2906,8 +2912,7 @@ class SYCLKernelNameTypeVisitor
     QualType T = TA.getAsType();
     if (const auto *ET = T->getAs<EnumType>())
       VisitEnumType(ET);
-    else
-      Visit(T);
+    Visit(T);
```

Comment:
When we stopped returning, we probably still need `else` branch here, so we don't go to visit all over again after we visited enum type.
```suggestion
  else
    Visit(T);
```
Or we can do early exit after we visited enum type
```
if (const auto *ET = T->getAs<EnumType>()) {
  VisitEnumType(ET);	      VisitEnumType(ET);
  return;  
}
Visit(T);
```